src/support/global.c

```diff
@@ -95,48 +188,32 @@ __global_calibrate_ticks(void)
 
 #if defined(__amd64) || defined(__aarch64__)
     {
-        struct timespec start, stop;
-        double ratio;
-        uint64_t diff_nsec, diff_tsc, min_nsec, min_tsc;
-        uint64_t tries, tsc_start, tsc_stop;
-        volatile uint64_t i;
-
-        /*
-         * Run this calibration loop a few times to make sure we get a reading that does not have a
-         * potential scheduling shift in it. The inner loop is CPU intensive but a scheduling change
-         * in the middle could throw off calculations. Take the minimum amount of time and compute
-         * the ratio.
-         */
-        min_nsec = min_tsc = UINT64_MAX;
-        for (tries = 0; tries < 3; ++tries) {
-            /* This needs to be CPU intensive and large enough. */
-            __wt_epoch(NULL, &start);
-            tsc_start = __wt_rdtsc();
-            for (i = 0; i < 100 * WT_MILLION; i++)
-                ;
-            tsc_stop = __wt_rdtsc();
-            __wt_epoch(NULL, &stop);
-            diff_nsec = WT_TIMEDIFF_NS(stop, start);
-            diff_tsc = tsc_stop - tsc_start;
-
-            /* If the clock didn't tick over, we don't have a sample. */
-            if (diff_nsec == 0 || diff_tsc == 0)
-                continue;
-            min_nsec = WT_MIN(min_nsec, diff_nsec);
-            min_tsc = WT_MIN(min_tsc, diff_tsc);
-        }
+        uint64_t epoch_ticks_min, epoch_ticks_avg, tsc_start, tsc_stop;
+        struct timespec clock_start, clock_stop;
 
-        /*
-         * Only use rdtsc if we got a good reading. One reason this might fail is that the system's
-         * clock granularity is not fine-grained enough.
-         */
-        if (min_nsec != UINT64_MAX) {
-            ratio = (double)min_tsc / (double)min_nsec;
-            if (ratio > DBL_EPSILON) {
-                __wt_process.tsc_nsec_ratio = ratio;
-                __wt_process.use_epochtime = false;
-            }
-        }
+        __get_epoch_call_ticks(&epoch_ticks_min, &epoch_ticks_avg);
+
+        if (!__get_epoch_and_ticks(&clock_start, &tsc_start, epoch_ticks_min, epoch_ticks_avg))
+            return;
+
+        __wt_sleep(0, CLOCK_CALIBRATE_USEC);
+
+        if (!__get_epoch_and_ticks(&clock_stop, &tsc_stop, epoch_ticks_min, epoch_ticks_avg))
+            return;
+
+        uint64_t diff_nsec = WT_TIMEDIFF_NS(clock_stop, clock_start);
+        uint64_t diff_tsc = tsc_stop - tsc_start;
+        if (diff_nsec < 10 || diff_tsc < 10)
```

Comment:
Is 10 also picked by experiment results on different platforms?

Reply:
Can we name the 10 something like `CLOCK_CALIBRATE_MIN_DIFF` or something?

Reply:
These constants are some "reasonable" ones. They can be reviewed later.

Macro: done.